lib/web/apiserver_test.go

```diff
@@ -352,24 +346,46 @@ func (s *WebSuite) authPack(c *C, user string) *authPack {
 	return &authPack{
 		otpSecret: otpSecret,
 		user:      user,
-		login:     login,
+		login:     s.user,
 		session:   sess,
 		clt:       clt,
 		cookies:   re.Cookies(),
 	}
 }
 
-func (s *WebSuite) createUser(c *C, user string, login string, pass string, otpSecret string) {
+// authPack returns authenticated client with default user role settings.
+func (s *WebSuite) authPack(c *C, user string) *authPack {
+	pass := "abc123"
+	rawSecret := "def456"
+	otpSecret := base32.StdEncoding.EncodeToString([]byte(rawSecret))
+
+	s.createUser(c, user, s.user, pass, otpSecret)
+
+	return s.authPackHelper(c, user, pass, otpSecret)
+}
+
+// extendedAuthPack is the same as authPack but the user role setting
+// is extended to more resources.
+func (s *WebSuite) extendedAuthPack(c *C, user string) *authPack {
```

Comment:
Instead of this:

```
extendedAuthPack
authPack
authPackHelper
createSuperUser
createUserHelper
createUser
```

Lets add a simple unit-test and _then_ we can think how to refactor existing setup.